to.etc.domui/src/main/java/to/etc/domui/util/exporters/ResultSetExporter.java

```diff
@@ -0,0 +1,110 @@
+package to.etc.domui.util.exporters;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.HorizontalAlignment;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+
+/**
+ * Basic exporter that dumps result set into workbook, as new sheet with specified name.
+ */
+@NonNullByDefault
+public class ResultSetExporter {
+
+	private final ExcelFormat m_format;
+
+	public ResultSetExporter(ExcelFormat format) {
+		m_format = format;
+	}
+
+	public void writeExcel(Workbook workbook, ResultSet rs, int rowCount, String sheetName) {
```

Comment:
We should not require the #of rows, because that always means doing a possibly expensive extra query. It is easy enough to just abort if you find that you are exceeding the row count while iterating over the dataset.

Reply:
yeah, that is true... I had it for free so I used it, but you are right - it would not always be there _for free_. I'll remove that.

Reply:
ok,should be fixed now